File: robitcubebot_controller/src/lifter_controller.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float64.hpp"

class LifterController : public rclcpp::Node
{
public:
  LifterController()
  : Node("lifter_controller")
  {
    publisher_ = this->create_publisher<std_msgs::msg::Float64>("/lift_joint/command", 10);
    timer_ = this->create_wall_timer(
      std::chrono::milliseconds(500),
      std::bind(&LifterController::timer_callback, this));
  }

private:
  void timer_callback()
  {
    auto message = std_msgs::msg::Float64();
    message.data = 0.01;  // Example velocity command
    publisher_->publish(message);
  }

  rclcpp::Publisher<std_msgs::msg::Float64>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<LifterController>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: robitcubebot_controller/robitcubebot_controller/lifter_controller.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float64

class LifterController(Node):
    def __init__(self):
        super().__init__('lifter_controller')
        self.publisher_ = self.create_publisher(Float64, '/lift_joint/command', 10)
        timer_period = 0.5  # seconds
        self.timer = self.create_timer(timer_period, self.timer_callback)
        self.i = 0

    def timer_callback(self):
        msg = Float64()
        msg.data = 0.01  # Example velocity command
        self.publisher_.publish(msg)
        self.i += 1

def main(args=None):
    rclpy.init(args=args)
    lifter_controller = LifterController()
    rclpy.spin(lifter_controller)
    lifter_controller.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
